### ISSUE TITLE
twisted.web.test.test_http.Expect100ContinueServerTests depends on dictionary iteration order

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6113]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6113|
|Type|enhancement|
|Created|2012-10-15 22:07:26Z|
|Branch|https://github.com/twisted/twisted/tree/continue-tests-better-6113|

Iteration over items in a dictionary is unspecified and unstable.  `Expect100ContinueServerTests` includes static strings which will only match actual output if dictionary iteration happens in a particular order.  This makes them fragile tests.  They should be changed not to depend on any particular order.

<details><summary>Searchable metadata</summary>

```
trac-id__6113 6113
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_continue_tests_better_6113 branches/continue-tests-better-6113
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__tests tests
time__1350338846000000 1350338846000000
changetime__1351945604000000 1351945604000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36172]) Branching to 'continue-tests-better-6113'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Split off from [#6110](https://github.com/twisted/twisted/issues/6110).  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/continue-tests-better-6113).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [36206]) Merge continue-tests-better-6113

Author: exarkun
Reviewer: itamarst
Fixes: [#6113](https://github.com/twisted/twisted/issues/6113)

Change the Twisted Web server tests for `100 Continue` responses so that they do not
depend on the particular ordering of headers in the response (thus making them not
depend on dictionary iteration order which is unstable).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

[#5757](https://github.com/twisted/twisted/issues/5757) was closed as a duplicate of this.
